File: java-code/ride.java
/*
ID: angela.14
LANG: JAVA
TASK: ride
*/
import java.io.*;

class ride{
    public static void main (String [] args) throws IOException {
        BufferedReader fin = new BufferedReader(new FileReader("ride.in"));
        PrintWriter fout = new PrintWriter(new BufferedWriter(new FileWriter("ride.out")));
        String line1 = fin.readLine();
        String line2 = fin.readLine();

        String result = null;
        if (convertInt(line1) % 47 == convertInt(line2) % 47) {
            result = "GO";
        }
        else {
            result = "STAY";
        }

        fout.println(result);
        fout.close();   

    }

    public static int convertInt (String line) {
        int product = 1;
        int charLen = line.length();
        for (int i=0; i<charLen; i++) {
            product = product * (line.charAt(i) - 'A'+1);
        }
        return product;
    }
}

File: 1.2 ride/ride.py
"""
ID: angela.14
LANG: PYTHON3
TASK: ride
"""
with open('ride.in', 'r') as fin:
    line1 = fin.readline().strip()
    line2 = fin.readline().strip()

def convertint(line):
    product = 1
    for i in line:
        product = product * (ord(i)-ord('A')+1)
    return product % 47

rst = 'GO\n' if convertint(line1) == convertint(line2) else 'STAY\n'
# if convertint(line1) == convertint(line2):
#     print("GO")
# else:
#     print("STAY")

with open('ride.out', 'w') as fout:
    fout.write(rst)